class.jetpack.php

```diff
@@ -6938,7 +6938,13 @@ public static function get_calypso_env() {
 	 * Will return true if a user has clicked to register, or is already connected.
 	 */
 	public static function jetpack_tos_agreed() {
-		return Jetpack_Options::get_option( 'tos_agreed' ) || self::is_active_and_not_development_mode();
+		$tos_agreed = Jetpack_Options::get_option( 'tos_agreed' ) || self::is_active_and_not_development_mode();
+
+		if ( $tos_agreed ) {
+			add_filter( 'jetpack_tos_agreed', '__return_true' );
```

Comment:
Are we sure that `Jetpack::jetpack_tos_agreed` would always fire before any tracking attempt is made?

The approach I was thinking we needed is akin to #13711 -- making the Status package more robust and moving more of those checks out of `class.jetpack.php`.

No matter the project consuming the Tracking package, we'd need to provide a package-based way to confirm TOS compliance. I didn't think it was a blocker since `Jetpack` class calls already exist in the package, but @enej makes a good point about there might be a difference in intent between `src` and `legacy`. (If so, we should document that better?)

Reply:
> Are we sure that Jetpack::jetpack_tos_agreed would always fire before any tracking attempt is made?

I'd think so since we check it quite early?
https://github.com/Automattic/jetpack/blob/e91efae878e1e068df303ca68d504aabe643fd0a/class.jetpack.php#L570

> No matter the project consuming the Tracking package, we'd need to provide a package-based way to confirm TOS compliance. 

Agreed, but I think we're far from that right now, not just with the Tracking package.